gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/launcher/ExecuteGobblinJobLauncher.java

```diff
@@ -84,13 +87,46 @@ public void submitJob(List<WorkUnit> workunits) {
       Config jobConfigWithOverrides = applyJobLauncherOverrides(ConfigUtils.propertiesToConfig(this.jobProps));
 
       Help.propagateGaaSFlowExecutionContext(this.jobProps);
-
-      EventSubmitterContext eventSubmitterContext = new EventSubmitterContext(this.eventSubmitter);
-
+      EventSubmitterContext eventSubmitterContext = new EventSubmitterContext(
+          addAdditionalMetadataTags(this.jobProps, this.eventSubmitter.getTags()),
+          eventSubmitter.getNamespace()
+      );
       int numWorkUnits = workflow.execute(ConfigUtils.configToProperties(jobConfigWithOverrides), eventSubmitterContext);
       log.info("FINISHED - ExecuteGobblinWorkflow.execute = {}", numWorkUnits);
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
+
+  private static List<Tag<?>> addAdditionalMetadataTags(Properties jobProps, List<Tag<?>> additionalTags) {
```

Comment:
if this be core gobblin-on-temporal stuff, how about a method of `EventSubmitterContext`?

e.g. call as:
```
EventSubmitterContext eventSubmitterContext = new EventSubmitterContext(this.eventSubmitter).withAugmentedMetadata(this.jobProps);
```
(`withAugmentedMetadata` would return a new `EventSubmitterContext` that's updated to have the new `Tag`s).